### PR TITLE
Log Analysis Example 

### PR DESCRIPTION
**Pull request by technovangelist**
At kubecon and other events and on discord, we have been asked how to analyse logs using ollama. This is a simple example of one approach to this.

### CODE REVIEW

File: examples/python-loganalysis/loganalysis.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import sys
+import re
+import requests
+import json
+
+prelines = 10
```
Comment (BruceMacD) [inline examples/python-loganalysis/loganalysis.py:6]:
```suggestion
# prelines and postlines represent the number of lines of context to include in the output around the error
prelines = 10
```

File: examples/python-loganalysis/loganalysis.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  error_lines = []
+  for i, line in enumerate(log_lines):
+    if re.search('error', line, re.IGNORECASE):
+      error_lines.append(i)
+
+  error_logs = []
+  for error_line in error_lines:
+    start_index = max(0, error_line - prelines)
+    end_index = min(len(log_lines), error_line + postlines)
+    error_logs.extend(log_lines[start_index:end_index])
```
Comment (BruceMacD) [inline examples/python-loganalysis/loganalysis.py:27]: Just a suggestion, I think you could do this all in one loop

```suggestion
error_logs = []
    for i, line in enumerate(log_lines):
        if "error" in line.lower():
            start_index = max(0, i - prelines)
            end_index = min(len(log_lines), i + postlines + 1)
            error_logs.extend(log_lines[start_index:end_index])
```

**Review by BruceMacD — COMMENTED** [review]
Cool example 👍

**Comment by technovangelist** [comment]
@BruceMacD any more thoughts?

**Review by BruceMacD — APPROVED** [review]
looks good to me, I could see an interesting blog post coming from this too

**Pull request merged**